services/netlify/netlify.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const { renderBuildStatusBadge } = require('../build-status')
+const { BaseSvgScrapingService } = require('..')
+
+const pendingStatus = 'building'
+const notBuiltStatus = 'not built'
+
+const statusMap = {
+  building: pendingStatus,
+  waiting: pendingStatus,
+  initiated: pendingStatus,
+  stopped: notBuiltStatus,
+  ignored: notBuiltStatus,
+  blocked: notBuiltStatus,
+  infrastructure_failure: 'failed',
+}
+
+module.exports = class Netlify extends BaseSvgScrapingService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'netlify',
+      pattern: ':projectId',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Netlify',
+        namedParams: {
+          projectId: 'e6d5a4e0-dee1-4261-833e-2f47f509c68f',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'passing' }),
+      },
+      // {
+      //   title: 'Netlify (branch)',
+      //   pattern: ':projectId/:branch',
+      //   namedParams: {
+      //     projectId: '0bdb0440-3af5-0133-00ea-0ebda3a33bf6',
+      //     branch: 'master',
+      //   },
+      //   staticPreview: renderBuildStatusBadge({ status: 'passing' }),
+      // },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'netlify',
+      namedLogo: 'netlify',
+    }
+  }
+
+  static render({ status }) {
+    status = statusMap[status] || status
+    const result = renderBuildStatusBadge({ status })
+    if (result.message === 'building' && !result.color) {
+      result.color = 'yellow'
+    }
+    return result
+  }
+
+  async fetch({ projectId, branch }) {
+    const url = `https://api.netlify.com/api/v1/badges/${projectId}/deploy-status`
+    const { buffer } = await this._request({
+      url,
+    })
+    if (buffer.includes('#EAFAF9')) return { message: 'passing' }
+    if (buffer.includes('#FFF3F4')) return { message: 'failing' }
+    if (buffer.includes('#FEFAEA')) return { message: 'building' }
+    return { message: 'project not found' }
```

Comment:
IIRC, a 404 response on the request to the above URL will automatically result in a `not found` badge being returned (I think this is why that not found test was failing before). You can also customize that not found message via something like `errorMessages: { 404: 'project not found' },` above alongside the `url` arg to `_request`. 

I'm not familiar with Netlify at all, but if we get down to this line here, is it really a case of the project not being found? The request would've had to have returned an HTTP 200 to get here, so just want to confirm this really is a project not found scenario and not more of an 'unknown' status. 

If it truly is a not found scenario, then I'd suggest throwing a NotFound error here instead of returning a 'project not found' status and then going through the render logic. To do that would be a:

update line 4 above to:
```js
const { BaseSvgScrapingService, NotFound } = require('..')
```

and then here line 76 would be:
```js
    throw new NotFound({ prettyMessage: 'project not found' })
```

Reply:
This should instead probably be:
```suggestion
    return { message: 'unknown' }
```

In case **Netlify** changes the colours of their badge icons.

Reply:
It turns out Netlify returns a JSON object instead of the SVG when a project is not found: https://api.netlify.com/api/v1/badges/not-found/deploy-status. Maybe we can use that for "not found" and "error" as the fallback.

Reply:
It also correctly returns HTTP `404` so @calebcartwright you're right, it automatically goes to the "not found" badge. Therefore the fallback should be "unknown" as @Exe-Boss suggested.